source/src/main/java/com/trustpilot/connector/dynamodb/utils/RecordConverter.java

```diff
@@ -145,4 +148,65 @@ private String sanitiseAttributeName(final String attributeName) {
 
         return sanitisedAttributeName;
     }
+
+    public static Struct getAttributeValueStruct(Map<String, AttributeValue> attributes) {        
+        final Struct attributeValueStruct = new Struct(getAttributeValueSchema(attributes));
+        
+        // Mapping dynamo db attributes to schema registry types (dynamo db attributes are documented at below link)
+        //https://github.com/aws/aws-sdk-java/blob/master/aws-java-sdk-dynamodb/src/main/java/com/amazonaws/services/dynamodbv2/model/AttributeValue.java
+
+        for (Map.Entry<String, AttributeValue> attribute : attributes.entrySet()) {
+            final String attributeName = attribute.getKey();
+            final AttributeValue attributeValue = attribute.getValue();
+            if (attributeValue.getS() != null) {
+                attributeValueStruct.put(attributeName, attributeValue.getS());
+            } else if (attributeValue.getN() != null) {
+                attributeValueStruct.put(attributeName, attributeValue.getN());
+            } else if (attributeValue.getB() != null) {
+                attributeValueStruct.put(attributeName, attributeValue.getB());
+            } else if (attributeValue.getSS() != null) {
+                attributeValueStruct.put(attributeName, attributeValue.getSS());
+            } else if (attributeValue.getNS() != null) {
+                attributeValueStruct.put(attributeName, attributeValue.getNS());
+            } else if (attributeValue.getBS() != null) {
+                attributeValueStruct.put(attributeName, attributeValue.getBS());
+            } else if (attributeValue.getNULL() != null) {
+                attributeValueStruct.put(attributeName, attributeValue.getNULL());
+            } else if (attributeValue.getBOOL() != null) {
+                attributeValueStruct.put(attributeName, attributeValue.getBOOL());
+            }           
+        }
+        return attributeValueStruct;
```

Comment:
should probably have an else?  what if none of the else ifs match?
Also -> i'm sure you recognize the ugliness.  Any better way?

Reply:
A switch maybe?

Reply:
yep i will refactor some of this code once i make sure this is working with data in dev. will make it more cleaner and easy to read with comments

Reply:
I looked at the underlying AttributeValue.java.  hideous code.
private String n; -> screw you!  
to be clear, not gurjit's code.